### PR TITLE
[12.0][FWD] #449 shopinvader: force update pricelist and prices

### PR DESCRIPTION
**Pull request by rousseldenis**

### CODE REVIEW

File: shopinvader/data/ir_cron.xml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+<odoo noupdate="1">
+    <record id="shopinvader_sale_order_price_recompute" model="ir.cron">
+        <field name="name">Shopinvader - Recompute prices on carts</field>
+        <field name="active" eval="True"/>
+        <field name="user_id" ref="base.user_root"/>
+        <field name="interval_number">1</field>
+        <field name="interval_type">days</field>
+        <field name="numbercall" eval="-1"/>
+        <field name="doall" eval="False"/>
+        <field name="model">shopinvader.backend</field>
```
Comment (simahawk) [inline shopinvader/data/ir_cron.xml]: :fire: you have to use `state` + `code` + `model_id` :wink:

**Review by sebastienbeau — APPROVED** [review]
/ocabot merge minor

File: shopinvader/models/shopinvader_backend.py [hunk 2]
```diff
@@ -435,3 +437,58 @@ def _normalize_name(name):
     def _onchange_name(self):
         if self.name and not self.tech_name:
             self.tech_name = self._normalize_name(self.name)
+        auth_api_key = getattr(request, "auth_api_key", None)
```
Comment (lmignon) [inline shopinvader/models/shopinvader_backend.py:440]: @sebastienbeau, @rousseldenis   PLZ REVERT !!!!!! "auth_api_key" should never be used here!

File: shopinvader/data/ir_cron.xml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    <record id="shopinvader_sale_order_price_recompute" model="ir.cron" forcecreate="True">
+        <field name="name">Shopinvader - Recompute prices on carts</field>
+        <field name="active" eval="True"/>
+        <field name="user_id" ref="base.user_root"/>
+        <field name="interval_number">1</field>
+        <field name="interval_type">days</field>
+        <field name="numbercall" eval="-1"/>
+        <field name="doall" eval="False"/>
+        <field name="model_id" ref="model_shopinvader_backend"/>
+        <field name="code">model._launch_sale_price_update()</field>
```
Comment (simahawk) [inline shopinvader/data/ir_cron.xml:14]: w/out setting  `state=code` this won't run at all

File: shopinvader/data/ir_cron.xml (hunk 1 above)
Comment (simahawk) [reply shopinvader/data/ir_cron.xml]: @rousseldenis ping :)

File: shopinvader/models/shopinvader_backend.py (hunk 2 above)
Comment (sebastienbeau) [reply shopinvader/models/shopinvader_backend.py:440]: Here is the fix : https://github.com/shopinvader/odoo-shopinvader/pull/738

**Pull request merged**